lib/json/named_declarations/parameter_declaration.ts

```diff
@@ -0,0 +1,30 @@
+import {ParameterDeclaration as tsParameterDeclaration} from 'typescript';
+
+import {convertExpression, convertTypeNode} from '../conversions';
+import {ConvertedSyntaxKind} from '../converted_syntax_kinds';
+import {Type} from '../types';
+
+import {NamedDeclaration} from './named_declaration';
+
+export class ParameterDeclaration extends NamedDeclaration {
+  private type: Type;
+  private optional = false;
+  private destructured = false;
+  private initializer?: string;
+
+  constructor(node: tsParameterDeclaration) {
+    super(node, ConvertedSyntaxKind.Parameter);
+
+    this.type = convertTypeNode(node.type);
+    if (node.questionToken) {
+      // TODO(derekx): Do we want to mark parameters with default values as optional?
```

Comment:
All parameters with default values are inherently optional, right? I can see the argument that it should only be marked optional if it's explicitly optional, but I don't see an instance in the conversion to Dart where that separation makes a difference.

Reply:
Seems like you went with the option to not mark them optional (which is fine, I'm just commenting to resolve this question).